src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -152,57 +159,8 @@ private function populateAvailableType($id, Definition $definition)
         }
 
         foreach ($definition->getAutowiringTypes() as $type) {
-            $this->definedTypes[$type] = true;
-            $this->types[$type] = $id;
-        }
-
-        if (!$reflectionClass = $this->getReflectionClass($id, $definition)) {
-            return;
-        }
-
-        foreach ($reflectionClass->getInterfaces() as $reflectionInterface) {
-            $this->set($reflectionInterface->name, $id);
+            $this->autowiringTypes[$type] = $id;
```

Comment:
Should we use the autowiring types in this feature ? Maybe should it be renamed then ?